ts/utils/messages.ts

```diff
@@ -95,6 +95,8 @@ type MessagePaymentUnexpirable = {
   amount: NonNullable<
     CreatedMessageWithContentAndAttachments["content"]["payment_data"]
   >["amount"];
+  expireStatus?: ExpireStatus;
```

Comment:
Since this logic is a little tricky could you:

- leave some comments to better understand what the code does?
- write some unit tests to test `paymentExpirationInfo` function?